src/sentry/models/release.py

```diff
@@ -214,6 +219,73 @@ def add_project(self, project):
         else:
             return True
 
+    def set_head_commits(self, head_commits, user, fetch_commits=False):
+        from sentry.models import Commit, ReleaseHeadCommit, Repository
+        from sentry.plugins import bindings
+
+        prev_release = type(self).objects.filter(
+            organization_id=self.organization_id,
+            projects__in=self.projects.all(),
+        ).order_by('-date_added').first()
+
+        commit_list = []
+
+        for head_commit in head_commits:
+            repo = Repository.objects.get_or_create(
+                organization_id=self.organization_id,
+                name=head_commit['repository'],
+            )[0]
+            commit = Commit.objects.get_or_create(
+                organization_id=self.organization_id,
+                repository_id=repo.id,
+                key=head_commit['current_id'],
+            )[0]
+            # update head commit for repo/release if exists
+            ReleaseHeadCommit.objects.create_or_update(
+                organization_id=self.organization_id,
+                repository_id=repo.id,
+                release=self,
+                values={
+                    'commit': commit,
+                }
+            )
+            if fetch_commits:
+                try:
+                    provider_cls = bindings.get('repository.provider').get(repo.provider)
+                except KeyError:
+                    continue
+
+                # if previous commit isn't provided, try to get from
+                # previous release otherwise, give up
+                if head_commit.get('previous_id'):
+                    start_sha = head_commit['previous_id']
+                elif prev_release:
+                    try:
+                        prev_head = ReleaseHeadCommit.objects.get(
+                            organization_id=self.organization_id,
+                            release=prev_release,
+                            repository_id=repo.id,
+                        )
+                    except ReleaseHeadCommit.DoesNotExist:
+                        continue
+                    start_sha = prev_head.commit.key
+                else:
+                    continue
+
+                end_sha = commit.key
+                provider = provider_cls(id=repo.provider)
+                try:
+                    reop_commits = provider.compare_commits(
+                        repo.name, start_sha, end_sha, actor=user
+                    )
+                except (PluginError, InvalidIdentity) as e:
```

Comment:
Yeah, I don't expect that we'd hit it, it's just behavior that could happen at some point becuase that's what the base plugin is doing. So it's expected, and we should handle it and ignore.

So more explicitly, I'm suggesting add an additional `except` block just for this and ignoring.